client/src/pages/OfferForm/index.tsx

```diff
@@ -0,0 +1,106 @@
+import { useState } from 'react'
+import './OfferForm.css'
+import { OfferData } from '../../types/offerTypes'
+
+interface OfferFormProps {
+  task: {
+    id: string
+    title: string
+  }
+  onSubmit: (offerData: OfferData) => void
+  onCancel: () => void
+}
+
+const OfferForm = ({ task, onSubmit, onCancel }: OfferFormProps) => {
+  const [offerData, setOfferData] = useState<OfferData>({
+    offer_description: '',
+    offer_media: null,
+    posting_time: '',
+    price: '',
+    offer_status: 'pending',
+  })
+
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => {
+    setOfferData({
+      ...offerData,
+      [event.target.name]: event.target.value,
+    })
```

Comment:
nit: you can use the set state with previous state btw, like:
```javascript
    setOfferData((prevData) => ({
      ...prevData,
      [name]: value,
    }));
```
both of them are fine though.